Sources/App/Features/Slots/Controllers/SlotViewController.swift

```diff
@@ -0,0 +1,81 @@
+import Vapor
+import Fluent
+
+struct SlotViewController: RouteCollection {
+
+    enum SlotTypes: String, CaseIterable {
+        case presentation, activity
+    }
+
+    private struct PresentationContext: Content {
+        let slot: Slot?
+        let events: [Event]
+        let presentations: [Presentation]
+        let activities: [Activity]
+        let initialType: String
+        var types = SlotTypes.allCases.map(\.rawValue)
+    }
+
+    func boot(routes: RoutesBuilder) throws {
+        routes.get("", use: onCreate)
+        routes.get(":id", use: onEdit)
+        routes.get("delete", ":id", use: onDelete)
+    }
+
+    private func onCreate(request: Request) async throws -> View {
+        guard request.user?.role == .admin else {
```

Comment:
Minor: an extension on user exists for this

Reply:
Ah my bad, I'll make sure it gets switched when doing the API endpoint for slots!